src/overlays/components/OverlayController.tsx

```diff
@@ -71,6 +71,7 @@ export const OverlayController: FunctionComponent = () => {
       case 'add-custom-token':
         activeOverlay.current = <CustomApiTokenOverlay onClose={onClose} />
         break
+      case 'access-cloned-token':
```

Comment:
since there is no `break` below the first case (`case 'access-cloned-token'`) it will continue to execute the next case statement. Both `access-cloned-token` and `access-token` will be using the same component -> `DisplayTokenOverlay`.

Reply:
I can write it in the normal `case 'blah': activeOverlay.current = <Component /> break` fashion, but i figured it would be repetitive since both cases will be rendering `DisplayTokenOverlay`.

Reply:
So I think this looks a bit weird to me and that's why it stands out in the review. What if we add
```
activeOverlay.current = <DisplayTokenOverlay />
        break
```
in the case, so we know exactly what it's returning?

Reply:
![Screen Shot 2022-01-18 at 4 24 45 PM](https://user-images.githubusercontent.com/66275100/150028442-80ae9df2-8787-4308-b926-7cc0a321ddbb.png)
Agreed, it might take a second to understand but Shmuel did the same thing with the `add note/edit note` case a year ago; I was lowkey following his lead.

Reply:
@blegesse-git @Palakp41 

Switch statement fall-throughs (which is what this is called, the statement is falling through to the next statement) are a very powerful and potentially dangerous pattern. Generally I'm of the mindset that it's fine on a small codebase shared between a couple of engineers that work closely together or in personal projects where you're the only person writing code.

I'm a little bit more leery of it in a case like ours where we have lots of code and engineers who don't interact super frequently (like Unity and Dumplings). There are a couple of things that make it pretty risky under those circumstances:

1. It's not explicit. Let's say you merged this in and Subir was looking at it in 3 months having never seen this conversation. It's not clear from context whether you meant to fall through or whether you made a mistake. He'd have to go `git blame` this line, find out when it got added, and then come read this conversation to understand the intent of the code.
2. It's potentially dangerous in the future when we got to change this code. It's much easier to make a mistake then, especially if the intent of the code isn't clear. The situation where we change this code is where the insidious and hard-to-track-down bugs get added.

Generally, if you're going to do something like this, you want to call attention to it with a comment explaining why. But if you wrote a comment explaining that it's falling through on purpose and the reason it's falling through is to avoid writing the same line in two different switch statements, I'm thinking, _why not just have a normal switch statement with a `break`_?

There are a couple of meta points I want to bring up as well:

- In larger codebases like ours, the intent of the code is usually harder to understand than recognizing duplication or repetition. It makes sense to optimize the code to make the intent clear and the code readable.
- Code spends the vast majority of its life being run and read, not written, so we should optimize it to make it easier to read, rather than easier to produce.
- In larger codebases shared among teams, repetition of code is usually much less detrimental than they teach you it is when you learn to program. It's not because repetitive code isn't necessarily a bad thing, it's that usually what you replace the repetition with is actually worse than the repetition itself. Bad abstractions, or potential bugs are way harder to work through than repetition. See [rule of three](https://understandlegacycode.com/blog/refactoring-rule-of-three/), which is somewhat orthogonal to this point, somewhat relevant, and still worth reading about.
- I tend to take a code reviewer being confused by code I wrote as a very strong signal that people maintaining this code will also be confused when reading it. A PR has a lot of context with it - there's a description, usually a ticket associated with it, and there are often screenshots and movies demonstrating it. If a reviewer still doesn't quite understand with all that extra context, it's probably a good indicator that the code can be made easier to follow, made simpler, or changed based on the information the reviewer is giving you.

So having written out that mini dissertation, I think it's best to take Palak's suggestion and just have a little bit of duplication and no fall-throughs. I'm approaching this as a learning / teaching moment, more than a "this is wrong and needs to be fixed moment." I'm trying to give you as much context into why I think that Palak's suggestion is the right one, not bury you under a mountain of "I'm right, you're wrong." 

So after that novel, what do you think of that?

Reply:
I admit I was putting too much emphasis on optimization and efficiency rather than readability. But more importantly, I saw the way the overlay's switch statement was structured and I thought it's best I follow lead. It almost didn't make sense to write it how its normally written when there is code few lines above written using the fall-through method. With that being said, I really appreciate you taking the time to break it down. 
The big takeaway from your comment is this; "Code spends the vast majority of its life being run and read, not written, so we should optimize it to make it easier to read, rather than easier to produce." I definitely wouldn't want the next person working on this to have to chase down github commits to understand the intent behind a code I wrote.

Reply:
while i'm at it, should I also update `activeOverlay.current = <NoteEditorOverlay onClose={onClose} />` to the `add note` case?

Reply:
@hoorayimhelping thanks for breaking it down so well! Very helpful to understand your thought process behind this.
@blegesse-git let's update that as well and make sure we test that overlay before we merge it in. Let me know if you have any questions

Reply:
@blegesse-git @Palakp41 I would prefer changing the other switch fall-through to be done in a follow-up PR, but it's not a hard requirement. It's a good practice to keep commits as logical, and singularly focused.

But at the same time I also think that practicality and pragmatism matter deeply and we shouldn't let perfect or ideal get in the way of better. So go ahead and make that change, it'll be fine 👍